### ISSUE TITLE
Bump drift version to ^2.18.0

### ISSUE THREAD

**Issue by priyanshu-sekhar** [aws-amplify/amplify-flutter#4845]
### Description

I am using amplify_storage_s3 ^1.8.0 in my flutter package. Tried adding the latest available version of drift package (^2.18.0) but am getting the following error - 
```
Because no versions of amplify_storage_s3 match >1.8.0 <2.0.0 and amplify_storage_s3 1.8.0 depends on amplify_storage_s3_dart ^0.3.11, amplify_storage_s3 ^1.8.0 requires amplify_storage_s3_dart ^0.3.11.
Because amplify_storage_s3_dart 0.3.11 depends on drift >=2.14.0 <2.15.0 and no versions of amplify_storage_s3_dart match >0.3.11 <0.4.0, amplify_storage_s3_dart ^0.3.11 requires drift >=2.14.0 <2.15.0.
Thus, amplify_storage_s3 ^1.8.0 requires drift >=2.14.0 <2.15.0.
```

### Categories

- [ ] Analytics
- [ ] API (REST)
- [ ] API (GraphQL)
- [ ] Auth
- [ ] Authenticator
- [ ] DataStore
- [ ] Notifications (Push)
- [X] Storage

### Steps to Reproduce

1. Create a new project
2. Add amplify_storage_s3: ^1.8.0 to pubspec.yaml
3. Add drift: ^2.18.0 to pubsec.yaml
4. Run flutter pub get

### Platforms

- [ ] iOS
- [ ] Android
- [X] Web
- [ ] macOS
- [ ] Windows
- [ ] Linux

### Flutter Version

3.19.3

### Amplify Flutter Version

1.8.0

### Deployment Method

Amplify CLI

**Comment by khatruong2009** [comment]
Hi @priyanshu-sekhar, we will look into this issue and get back to you when we have an update.

**Comment by khatruong2009** [comment]
Hi @priyanshu-sekhar, we have the upper bound of our drift dependency set to 2.15.0. Anything at or above 2.14.0 and below 2.15.0 will work. Can you try setting your drift version to 2.14.0?

**Comment by priyanshu-sekhar** [comment]
@khatruong2009 Yes I have updated my specs to use ^2.14.1 for now, but I wanted to use some functionalities introduced in drift ^2.16.0 and above. Is there a plan to upgrade the dependency constraints ?

**Comment by khatruong2009** [comment]
Hi @priyanshu-sekhar, we are working to upgrade the drift dependency. You can follow the progress in the PR mentioned above.

**Comment by priyanshu-sekhar** [comment]
Perfect, thanks! Will keep an eye on this 👍

**Comment by Aanu1995** [comment]
Any update on this. I am only using AWS Cognito and this is affecting me

**Comment by Equartey** [comment]
@priyanshu-sekhar @Aanu1995 
Hi all, this has been addressed in Amplify Flutter 2.3.0. Please let us know if you have any more concerns.

**Comment by priyanshu-sekhar** [comment]
Thanks team, this solves a lot of issues for us now.
